File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count , n , element = 1, len(nums), nums[0]
        for i in range(1, n):
            if count is 0:
                element = nums[i]
                count += 1
            elif nums[i] is element:
                count += 1
            else:
                count -= 1
        return element

File: LeetCode/0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 1, n = nums.length, element = nums[0];
        for (int i = 1; i < n; i++) {
            if (count == 0) {
                element = nums[i];
                count++;
            } else if (nums[i] == element) {
                count++;
            } else {
                count--;
            }
        }
        return element;
    }
}